### PR TITLE
UI: Dex integration

### PR DESCRIPTION
**Pull request by carlito-scality**
**Component**: UI 

**Context**: 
- Integrate Dex login using redux-oidc library

**Summary**:

**Acceptance criteria**: 
- The MetalK8s UI should directly use dex to authenticate a user using OAuth2 flow.
- Once logged in, the reste should works like before without regressions

---

Closes: https://github.com/scality/metalk8s/issues/2015

### CODE REVIEW

**Comment by gdemonet** [comment]
@carlito-scality you'll need to rebase on `development/2.5` when #2063 gets merged.

File: salt/metalk8s/addons/ui/deployed/ui.sls [hunk 1]
```diff
@@ -1,6 +1,12 @@
 include:
 - .namespace
 
+{%- set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{%- set context = "kubernetes-admin@kubernetes" %}
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/ui/deployed/ui.sls]: This is no longer needed, please remove it

**Review by ChengYanJin — DISMISSED** [review]
LGTM! Great job!!!
We will need to handle share user data between tabs in order to eliminate unnecessary authentication.

**Review by ChengYanJin — DISMISSED** [review]

**Review by TeddyAndrieux — APPROVED** [review]
backend part LGTM

**Pull request merged**